Sources/StreamChatUI/ChatThread/ChatThreadVC.swift

```diff
@@ -0,0 +1,227 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+/// Controller responsible for displaying message thread.
+@available(iOSApplicationExtension, unavailable)
+open class ChatThreadVC: _ViewController, ThemeProvider {
+    /// Controller for observing data changes within the channel
+    open var channelController: ChatChannelController!
+
+    /// Controller for observing data changes within the parent thread message.
+    open var messageController: ChatMessageController!
+
+    public var client: ChatClient {
+        channelController.client
+    }
+
+    /// Component responsible for setting the correct offset when keyboard frame is changed
+    open lazy var keyboardHandler: KeyboardHandler = ComposerKeyboardHandler(
+        composerParentVC: self,
+        composerBottomConstraint: messageComposerBottomConstraint
+    )
+
+    /// User search controller passed directly to the composer
+    open lazy var userSuggestionSearchController: ChatUserSearchController =
+        channelController.client.userSearchController()
+
+    open lazy var messageListVC: ChatMessageListVC = components
```

Comment:
in 3 line above there is usage of:
>    /// Component responsible for setting the correct offset when keyboard frame is changed
    open lazy var keyboardHandler: KeyboardHandler = ComposerKeyboardHandler(
        composerParentVC: self,
        composerBottomConstraint: messageComposerBottomConstraint
    )
    
could we think about removing short `.init` statements, they are decreasing readability, cause you won't see exact type

Reply:
There is no exact type because customers can set their own 🙂

Reply:
we could try to directly state that this is a injected Type we are using like 
`components.messageComposerVC.Type()` or something with full-type-name constructor

Reply:
The exact type is on the property of the variable (`messageListVC: ChatMessageListVC`), and it is actually a rule of the SDK when declaring properties on the UI components, you need to explicitly declare the type. But the customer can override this, or set a subclass of the component, so we need to use `init()`. `components.messageComposerVC.init()` vs `components.messageComposerVC.Type()` is a minor detailed IMO, but we can discuss later in the team if you feel it is worth it 👍

Reply:
lets put on TBD list, readability always worth the discussion :)